### PR TITLE
fix: ROA maxLength bounds checks

### PR DESCRIPTION
**Pull request by natesales**

### CODE REVIEW

File: validator/lib/roa.go [hunk 1]
```diff
@@ -154,6 +154,21 @@ func (entry *ROAEntry) Validate() error {
 	if entry.MaxLength < s {
 		return errors.New(fmt.Sprintf("Max length (%v) is smaller than prefix length (%v)", entry.MaxLength, s))
 	}
+
+	if entry.MaxLength < 1 {
+		return fmt.Errorf("max length (%d) is less than 1", entry.MaxLength)
```
Comment (lspgn) [inline validator/lib/roa.go:159]: Nit: in theory one could have 0.0.0.0/0 with max length of 0

File: validator/lib/roa.go [hunk 2]
```diff
@@ -154,6 +154,21 @@ func (entry *ROAEntry) Validate() error {
...
+	}
+
+	if entry.IPNet.IP.To4() != nil { // If IPv4
+		if entry.MaxLength > 32 {
```
Comment (lspgn) [inline validator/lib/roa.go:163]: You could pack the nested if on the same line

**Comment by lspgn** [comment]
LGTM

**Review by lspgn — APPROVED** [review]

**Comment by ties** [comment]
Out of curiosity: What happens with a non-zero host part? (e.g. `203.0.113.128/24`)

**Review by leev — APPROVED** [review]

**Comment by nskalis** [comment]
thank you guys for taking care this scenario. do you know more or less when you expect to cut a new release (with this fix included) ?

**Comment by dhaynespls** [comment]
@nskalis https://github.com/cloudflare/cfrpki/releases/tag/v1.3.0 :D

**Pull request merged**